File: sdkwork-ai-sdk-typescript/src/types/knowledge-search-request.ts
import type { Filter } from './filter';

/** Search request */
export interface KnowledgeSearchRequest {
  /** 查询文本 */
  query: string;
  /** 过滤条件 */
  filters?: Filter[];
  /** 知识库ID */
  knowledge_base_id: string;
  /** 返回结果数量 */
  top_k?: number;
  /** 相似度阈值 */
  score_threshold?: number;
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/models/knowledge_search_request.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class KnowledgeSearchRequest:
    """Search request"""
    query: str
    filters: List[Filter] = None
    knowledge_base_id: str
    top_k: int = None
    score_threshold: float = None
